### PR TITLE
`Development`: Add Aeolus build plan templates for C programming exercises

### PR DESCRIPTION
**Pull request by reschandreas**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).
#### Server
- [x] **Important**: I implemented the changes with a very good performance and prevented too many (unnecessary) database calls.
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/).
#### Changes affecting Programming Exercises
- [x] I tested **all** changes and their related features with **all** corresponding user types on Test Server 1 (Atlassian Suite).

### Motivation and Context

This PR adds the templates for build plans for the language C. 

### Description

I added the templates and added the `ProgrammingLanguageFeature` for LocalCI so one can now create custom build plans (starting with the help of the template) for C in LocalCI and Bamboo.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Students
- Programming Exercise Setup with either LocalCI or Bamboo

1. Log in to Artemis
2. Create a new C exercise and enable custom build plans
3. Confirm that the build plan behaves the same as it does in Bamboo. meaning that base fails and solution hopefully passes

#### Exam Mode Testing (should behave exactly the same, same code is used)

Prerequisites:
- 1 Instructor
- 2 Students
- 1 Exam with a Programming Exercise

1. Log in to Artemis
2. Create an exam
3. Create a new C exercise and enable custom build plans
4. Confirm that the build plan behaves the same as it does in Bamboo. meaning that base fails and solution hopefully passes

#### Performance Review
- [ ] I (as a reviewer) confirm that the client changes (in particular related to REST calls and UI responsiveness) are implemented with a very good performance 
- [x] I (as a reviewer) confirm that the server changes (in particular related to database calls) are implemented with a very good performance
#### Code Review
- [x] Code Review 1
- [ ] Code Review 2
#### Manual Tests
- [x] Test 1
- [ ] Test 2
#### Exam Mode Test
- [ ] Test 1
- [ ] Test 2

### CODE REVIEW

**Review by milljoniaer — APPROVED** [review]
Demonstrated in testing session, works as expected 👍

**Review by laurenzfb — APPROVED** [review]
demonstrated in testing session. GCC and TACT were built correctly

**Review by milljoniaer — DISMISSED** [review]
Reapprove after solving merge conflicts

**Review by laurenzfb — APPROVED** [review]
reapprove after resolving merge conflicts

**Review by vinceclifford — DISMISSED** [review]
Works as expected

File: src/main/java/de/tum/in/www1/artemis/service/connectors/aeolus/AeolusTemplateService.java [hunk 1]
```diff
@@ -39,6 +38,8 @@ public class AeolusTemplateService {
 
     private final ResourceLoaderService resourceLoaderService;
 
+    private final HashMap<String, Windfile> templateCache = new HashMap<>();
```
Comment (basak-akan) [inline src/main/java/de/tum/in/www1/artemis/service/connectors/aeolus/AeolusTemplateService.java:41]: Maybe replacing HashMap with ConcurrentHashMap would be a better idea for thread safety

**Review by basak-akan — COMMENTED** [review]
Code looks good I just have one comment

File: src/main/java/de/tum/in/www1/artemis/service/connectors/aeolus/AeolusTemplateService.java (hunk 1 above)
Comment (reschandreas) [reply src/main/java/de/tum/in/www1/artemis/service/connectors/aeolus/AeolusTemplateService.java:41]: you're absolutely right, thanks!

**Review by basak-akan — APPROVED** [review]
Thank you for addressing my comment, code looks good

**Review by vinceclifford — APPROVED** [review]
Reapproval after code change

**Pull request merged**